search/fibonacci_search.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @function fibonacciSearch
+ * @description Fibonacci search algorithm for a sorted array.
+ *
+ * The algorithm searches for a specific value in a sorted array using Fibonacci numbers
+ * to divide the array into smaller subarrays. This algorithm is useful for large arrays where
+ * the cost of accessing elements is high.
+ *
+ * @param {number[]} array - sorted list of numbers
+ * @param {number} target - target number to search for
+ * @return {number} - index of the target number in the list, or -1 if not found
+ * @see [FibonacciSearch](https://www.geeksforgeeks.org/fibonacci-search/)
+ * @example fibonacciSearch([1,2,3], 2) => 1
+ * @example fibonacciSearch([4,5,6], 2) => -1
+ */
+
+export const fibonacciSearch = (array: number[], target: number): number => {
+    const arrayLength = array.length;
+    let fibNMinusTwo = 0;  // (n-2)'th Fibonacci No.
```

Comment:
These names are a bit unwieldy. I think it would be fine to call them something short like `a, b, c` given the comments.

Reply:
It was more understandable to me with self-explaining variables. However, they could cause more confusion than anything else. I changed them.